testcase/ninja_symlink_outputs.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+#
+# Copyright 2020 Google Inc. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+mk="$@"
+
+cat <<EOF >Makefile
+dangling_symlink:
+	ln -sf nil dangling_symlink
+dangling_symlink: .KATI_SYMLINK_OUTPUTS := dangling_symlink
+
+file:
+	touch file
+
+resolved_symlink: file
+	ln -sf file resolved_symlink
+resolved_symlink: .KATI_SYMLINK_OUTPUTS := resolved_symlink
+
+incorrectly_declared_symlink:
+	ln -sf nil incorrectly_declared_symlink
+incorrectly_declared_symlink: .KATI_SYMLINK_OUTPUTS := something_else
+
+foo bar: file
+	ln -sf file foo && cp foo bar
+foo: .KATI_SYMLINK_OUTPUTS := foo
+bar: .KATI_SYMLINK_OUTPUTS := bar
+EOF
+
+all="dangling_symlink resolved_symlink foo bar"
+if echo "${mk}" | grep kati > /dev/null; then
+  mk="${mk} --use_ninja_symlink_outputs"
+fi
+${mk} -j1 $all
+
+if [ -e ninja.sh ]; then
+  ./ninja.sh -j1 $all
+
+  if ! grep -A1 "build dangling_symlink:" build.ninja | grep -q "symlink_outputs = dangling_symlink"; then
```

Comment:
wow. `-A1` is a nice trick :) (it's a little brittle though because it relies on the `symlink_outputs` variable being the first)